src/app/_utils/getDigestArticleData.ts

```diff
@@ -1,24 +1,52 @@
+import removeMarkdown from 'remove-markdown'
+
 import { PATHS } from '@/constants/paths'
 
-import { convertMarkdownToDigestArticleData } from '@/utils/convertMarkdownToDigestArticleData'
-import { getAllData, getData } from '@/utils/getData'
+import { getAllMarkdownData, getMarkdownData } from '@/utils/getMarkdownData'
+
+import { DigestArticleFrontMatterSchema } from '@/schemas/DigestArticleFrontMatterSchema'
 
 import { sortArticlesByNumber } from '@/digest/utils/sortArticlesByNumber'
 
 const DIGEST_DIRECTORY_PATH = PATHS.DIGEST.entriesContentPath as string
+const CONTENT_PREVIEW_CHARACTER_LENGTH = 220
 
 export function getDigestArticleData(slug: string) {
-  return getData(
-    DIGEST_DIRECTORY_PATH,
-    convertMarkdownToDigestArticleData,
+  return getMarkdownData({
     slug,
-  )
+    directoryPath: DIGEST_DIRECTORY_PATH,
+    zodParser: DigestArticleFrontMatterSchema.parse,
+  })
 }
 
-export function getDigestArticlesData() {
-  return getAllData(DIGEST_DIRECTORY_PATH, convertMarkdownToDigestArticleData)
+export function getAllDigestArticleData() {
+  const articles = getAllMarkdownData({
+    directoryPath: DIGEST_DIRECTORY_PATH,
+    zodParser: DigestArticleFrontMatterSchema.parse,
+  })
+
+  return articles.map((article) => ({
+    ...article,
+    description: generatePreviewDescription(article.content),
```

Comment:
Can we use Zod for this?

Reply:
doesn't `truncateText` function already ensure the maximum length constraint?

Reply:
Agree for the naming 👍